speech/src/main/java/com/example/speech/InfiniteStreamRecognize.java

```diff
@@ -93,81 +95,19 @@ public static String convertMillisToDate(double milliSeconds) {
                 - TimeUnit.MINUTES.toSeconds(TimeUnit.MILLISECONDS.toMinutes(millis))));
   }
 
-  /** Performs infinite streaming speech recognition */
   public static void infiniteStreamingRecognize(String languageCode) throws Exception {
+    infiniteStreamingRecognize(languageCode, new MicBuffer(), getResponseObserver());
+  }
```

Comment:
What is a value of using these two additional parameters instead of initiating them inside `infiniteStreamingRecognize`?
The current version makes the code sample more complicated to understand.
Please, refactor the code to remove unnecessary nested calls and initializations.
If your plan is to demonstrate both a creation of the response observer **AND** stream recognition implementation, consider creating a separate sample for the response observer in a stand alone file and reuse it in your code sample of stream recognition.